src/repositories/powerSnapshotRepository.ts

```diff
@@ -0,0 +1,37 @@
+import { PowerSnapshot } from '../entities/powerSnapshot';
+import { Project } from '../entities/project';
+
+export const findInCompletePowerSnapShots = async (): Promise<
+  PowerSnapshot[]
+> => {
+  return PowerSnapshot.createQueryBuilder()
+    .where('"blockNumber" IS NULL')
+    .getMany();
+};
+
+export const findPowerSnapshotById = async (
+  id: number,
+): Promise<PowerSnapshot | undefined> => {
+  return PowerSnapshot.createQueryBuilder()
+    .where('id=:id', {
+      id,
+    })
+    .getOne();
+};
+
+export const updatePowerSnapShots = async (params: {
```

Comment:
@mohammadranjbarz if you agree with the other comment, we can remote this method